forta-bots/L1-Teleport-Backing/src/fetchAPI.ts

```diff
@@ -0,0 +1,83 @@
+import axios from "axios";
+import { getEndDate } from "./utils";
+
+const FORTA_API_ENDPOINT: string = "https://api.forta.network/graphql";
+
+export default class Fetcher {
+  readonly endpoint: string;
+  private post: any;
+
+  constructor(endpoint: string = FORTA_API_ENDPOINT, post: any = axios.post) {
+    this.post = post;
+    this.endpoint = endpoint;
+  }
+
+  public async L2HashGUIDExists(networkId: number, blockTimestamp: number, hash: string): Promise<boolean> {
+    let offset: any = {};
+    let hashesArray: string[] = [];
+
+    while (true) {
+      const { data } = await this.post(
+        this.endpoint,
+        {
+          query: QUERY,
+          variables: queryInput(networkId, blockTimestamp, offset),
+        },
+        {
+          headers: {
+            "content-type": "application/json",
+          },
+        }
+      );
+
+      for (let alert of data.data.alerts.alerts) {
+        // @ts-ignore
+        hashesArray.push(...Object.values(alert.metadata));
+      }
+
+      if (hashesArray.includes(hash)) {
+        return true;
+      }
```

Comment:
Lets get rid of `hashesArray` by doing this:
```suggestion
      for (let alert of data.data.alerts.alerts) {
        if(Object.values(alert.metadata).includes(hash)) {
          return true;
        }
      }
```